tools/clang/include/clang/AST/Type.h

```diff
@@ -2832,6 +2834,138 @@ class ExtVectorType : public VectorType {
   }
 };
 
+/// Represents a matrix type for HLSL matrix type, where "rows" specifies
+/// number of rows and "columns" specifies the number of columns.
+class MatrixType : public Type, public llvm::FoldingSetNode {
+protected:
+  friend class ASTContext;
+
+  /// The element type of the matrix.
+  QualType ElementType;
+
+  MatrixType(TypeClass TypeClass, QualType ElementTy, QualType CanonElementTy,
+             const Expr *RowExpr = nullptr, const Expr *ColumnExpr = nullptr);
+
+public:
+  /// Returns type of the elements being stored in the matrix
+  QualType getElementType() const { return ElementType; }
+
+  /// Valid elements types are the following:
+  /// * an integer type (as in C2x 6.2.5p19), but excluding enumerated types
+  ///   and _Bool
```

Comment:
HLSL allows the HLSL `bool` type in matrix though.  Is this just an issue with the comment, or the code as well?

Reply:
Ok, looks like `isRealType()` includes `bool` in this code base anyway.